default-with-analytics/WSO2AM210-All-In-One-Deployment-VPC.template.yaml

```diff
@@ -235,126 +262,94 @@ Resources:
       KeyName: !Ref KeyPairName
       SecurityGroups:
         - !Ref WSO2APIMSecurityGroup
-      IamInstanceProfile: !Ref WSO2APIMAWSCLIInstanceProfile
-      UserData: !Base64 
+      UserData: !Base64
         'Fn::Join':
           - |+
 
           - - '#!/bin/bash'
             - set -e
             - 'export PATH=~/.local/bin:$PATH'
-            - !Join 
-              - ''
-              - - sed -i "s/CF_ENI_ID/
-                - !Ref WSO2APIMNode1NetworkInterface
-                - /g" /usr/local/bin/allocate_ips.sh
-            - !Join 
-              - ''
-              - - sed -i "s/CF_ALLOC_ID/
-                - !GetAtt 
-                  - WSO2APIMNode1EIP
-                  - AllocationId
-                - /g" /usr/local/bin/allocate_ips.sh
-            - >-
-              sed -i "s/CF_DEFAULT_GATEWAY/10.0.254.1/g"
-              /usr/local/bin/allocate_ips.sh
-            - !Join 
-              - ''
-              - - sed -i "s/CF_ENI_PRIVATE_IP/
-                - !GetAtt 
-                  - WSO2APIMNode1NetworkInterface
-                  - PrimaryPrivateIpAddress
-                - /g" /usr/local/bin/allocate_ips.sh
             - mkdir -p ~/.aws
             - cat > ~/.aws/config << EOF
             - '[default]'
-            - !Join 
+            - !Join
               - ''
               - - 'region = '
                 - !Ref 'AWS::Region'
             - output = text
             - EOF
-            - bash /usr/local/bin/allocate_ips.sh
             - mkdir -p /mnt/efs
-            - !Join 
-              - ''
-              - - echo "
-                - !Ref WSO2APIMEFSFileSystem
-                - .efs.
-                - !Ref 'AWS::Region'
-                - .amazonaws.com
-                - >-
-                  :/ /mnt/efs nfs4
-                  nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2
-                  0 0" >> /etc/fstab
-            - !Join 
+            - apt-get update
+            - apt-get install -y nfs-common
```

Comment:
Wouldn't it be better to move these in to the AMI creation part? It would,

1. reduce time taken to start instances
2. actually be required by AMI validation process by AWS

Reply:
If we run these during AMI creation, at the point of running this AMI within an EC2 instance it will not acquire the latest package updates. As a result, I have added these to UserData section of the launch configurations, to be run at the point of running the AMI in an EC2 instance.

Reply:
Makes sense. I think in the future, we'd have to find a compromise between updates vs stability.